src/sentry/workflow_engine/models/incident_groupopenperiod.py

```diff
@@ -16,8 +19,20 @@ class IncidentGroupOpenPeriod(DefaultFieldsModel):
     __relocation_scope__ = RelocationScope.Excluded
 
     incident_id = BoundedBigIntegerField(null=True, unique=True)
+    incident_identifier = models.IntegerField(null=True)
     group_open_period = FlexibleForeignKey("sentry.GroupOpenPeriod", unique=True)
 
     class Meta:
         db_table = "workflow_engine_incidentgroupopenperiod"
         app_label = "workflow_engine"
+        constraints = [
+            models.CheckConstraint(
+                condition=Q(incident_identifier__isnull=False) & Q(incident_id__isnull=False)
+                | Q(incident_identifier__isnull=True) & Q(incident_id__isnull=True),
+                name="inc_id_inc_identifier_together",
+            ),
+            models.UniqueConstraint(
+                fields=["incident_id", "group_open_period"],
+                name="workflow_engine_uniq_incident_groupopenperiod",
+            ),
```

Comment:
We don't need this - there's already a unique on `incident_id` that is more specific, since  each incident id can only exist once in the table. Same for `group_open_period`